vehicle.go

```diff
@@ -0,0 +1,164 @@
+		return errors.New("unit must either be metric or imperial")
+	}
+	vehicle.UnitSystem = unit
+	return nil
+}
+
+// Info sends request to Smartcar API vehicle/ endpoint and returns formatted response.
+func (vehicle *Vehicle) Info() (VehicleInfoResponse, error) {
+	res, err := vehicle.request("/", requests.GET, nil)
+	if err != nil {
+		return VehicleInfoResponse{}, err
+	}
+
+	formattedResponse := new(VehicleInfoResponse)
+	defer res.Body.Close()
+	fmtErr := requests.FormatResponse(res.Body, formattedResponse)
+	if fmtErr != nil {
+		return VehicleInfoResponse{}, fmtErr
+	}
+
+	return *formattedResponse, nil
+}
+
+// VIN sends request to Smartcar API vehicle/vin endpoint and returns formatted response.
+func (vehicle *Vehicle) VIN() (string, error) {
+	res, resErr := vehicle.request("/vin", requests.GET, nil)
+	if resErr != nil {
+		return "", resErr
+	}
+
+	formattedResponse := new(struct{ Vin string })
+	defer res.Body.Close()
+	fmtErr := requests.FormatResponse(res.Body, formattedResponse)
+	if fmtErr != nil {
+		return "", fmtErr
+	}
+
+	return formattedResponse.Vin, nil
+}
+
+// Lock sends request to Smartcar API vehicle/odometer endpoint and returns formatted response.
+func (vehicle *Vehicle) Lock() (VehicleResponse, error) {
+	return vehicle.security("LOCK")
+}
+
+// Unlock sends request to Smartcar API vehicle/odometer endpoint and returns formatted response.
+func (vehicle *Vehicle) Unlock() (VehicleResponse, error) {
+	return vehicle.security("UNLOCK")
+}
+
+// Internal function to send lock/unlock request
+func (vehicle *Vehicle) security(action string) (VehicleResponse, error) {
+	var jsonStr []byte
+	if action == "LOCK" {
+		jsonStr = []byte(`{"action":"LOCK"}`)
+	} else {
+		jsonStr = []byte(`{"action":"UNLOCK"}`)
+	}
+
+	body := bytes.NewBuffer(jsonStr)
+
+	res, err := vehicle.request("/security", requests.POST, body)
+	if err != nil {
+		return VehicleResponse{}, err
+	}
+
+	formattedResponse := new(VehicleResponse)
+	defer res.Body.Close()
+	fmtErr := requests.FormatResponse(res.Body, formattedResponse)
+	if fmtErr != nil {
+		return VehicleResponse{}, fmtErr
+	}
+
+	return *formattedResponse, nil
+}
+
+// Permissions sends request to Smartcar API vehicle/permissions endpoint and returns formatted response.
+func (vehicle *Vehicle) Permissions() ([]string, error) {
+	res, err := vehicle.request("/permissions", requests.GET, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	formattedResponse := new(struct{ Permissions []string })
+	defer res.Body.Close()
+	fmtErr := requests.FormatResponse(res.Body, formattedResponse)
+	if fmtErr != nil {
+		return nil, fmtErr
+	}
+
+	return formattedResponse.Permissions, nil
+}
+
+// HasPermissions checks if vehicle has the permissions passed in.
+func (vehicle *Vehicle) HasPermissions(permissions ...string) (bool, error) {
+	vehiclePermissions, err := vehicle.Permissions()
+	if err != nil {
+		return false, err
+	}
+
+	set := make(map[string]bool)
```

Comment:
It's a little safer to use `map[string]struct{}`, since the behavior is ambiguous if there is a `false` in the map. I just googled it and found [this example](https://www.davidkaya.com/sets-in-golang/), there may be a better one out there though :)